fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CTASAnalyzer.java

```diff
@@ -44,43 +40,24 @@ public CTASAnalyzer(Catalog catalog, ConnectContext session) {
     @Deprecated
     public Relation transformCTASStmt(CreateTableAsSelectStmt createTableAsSelectStmt) {
         List<String> columnNames = createTableAsSelectStmt.getColumnNames();
-        QueryStmt queryStmt = createTableAsSelectStmt.getQueryStmt();
+        QueryStatement queryStatement = createTableAsSelectStmt.getQueryStatement();
         CreateTableStmt createTableStmt = createTableAsSelectStmt.getCreateTableStmt();
 
         if (createTableStmt.isExternal()) {
             throw new SemanticException("CTAS does not support create external table");
         }
 
-        QueryRelation queryRelation = new QueryAnalyzer(catalog, session)
-                .transformQueryStmt(queryStmt, new Scope(RelationId.anonymous(), new RelationFields()));
+        new Analyzer(catalog, session).analyze(queryStatement);
 
         // Pair<TableName, Pair<ColumnName, ColumnAlias>>
         Map<Pair<String, Pair<String, String>>, Table> columnNameToTable = Maps.newHashMap();
         Map<String, Table> tableRefToTable = Maps.newHashMap();
+        new AnalyzerUtils.TableCollector(tableRefToTable, session).visit(queryStatement);
 
         // For replication_num, we select the maximum value of all tables replication_num
         int defaultReplicationNum = 1;
-        List<TableRef> tableRefs = ((SelectStmt) queryStmt).getTableRefs();
```

Comment:
Because the newly parsed Statement has replaced the original TableRef with Relation, correspondingly, this logic has been replaced by AnalyzerUtils.TableCollector